study1/week3/reduce-array-size-to-the-half.java

```diff
@@ -0,0 +1,44 @@
+//https://leetcode.com/problems/reduce-array-size-to-the-half/description/
+
+class Solution {
+    public int minSetSize(int[] arr) {
+        //time complexity is O(n)
+        //space complexity is also O(n)
+        
+        int arrLength = arr.length;
+        int result = 0;
+
+        Map<Integer,Integer> map = new HashMap<Integer,Integer>();
+
+        //set up the frequency map
+        for (int value : arr) {
+            //if (map.containsKey(value)) {
+            //    map.put(value, map.getOrDefault(value, 0) + 1);
+            //}
+            //else {
+            //    map.put(value, 0);
+            //}
+            map.put(value, map.getOrDefault(value, 0) + 1);
+
+        }
+        
+        // Sort the map entries by frequency in descending order
+        List<Map.Entry<Integer, Integer>> entryList = new ArrayList<>(map.entrySet());
```

Comment:
After counting the numbers, the key actually becomes irrelevant, as we only need the count, so you can make this a simple List with one Integer type in it, so :
```java
List<Integer> entryList = new ArrayList<>(map.values());
entryList.sort();
```

but as the sort() sort it as ascending order, you need to iterate backwards.